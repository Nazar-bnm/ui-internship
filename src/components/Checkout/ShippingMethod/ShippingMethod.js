import React from 'react';
import cx from 'classnames';

import { Button } from '@/shared';

import './ShippingMethod.scss';

const CN = 'shipping-method';

const ShippingMethod = () => (
  <div className={cx(CN)}>
    <p className={`${CN}__text`}>Please choose a shipping method to deliver your order:</p>
    <form className={`${CN}__radiobtns-form`}>
      <div className={`${CN}__radiobtn`}>
        <input
          className={`${CN}__radiobtn-input`}
          type="radio"
          name="ship-to-this-address"
          value="ship-to-this-address"
        />
        <label htmlFor="ship-to-this-address">UPS (ground) $7.25</label>
      </div>
      <div className={`${CN}__radiobtn`}>
        <input
          className={`${CN}__radiobtn-input`}
          type="radio"
          name="ship-to-this-address"
          value="ship-to-different-address"
        />
        <label htmlFor="guest">UPS (7 days select) $9.75</label>
      </div>
      <div className={`${CN}__radiobtn`}>
        <input
          className={`${CN}__radiobtn-input`}
          type="radio"
          name="ship-to-this-address"
          value="ship-to-different-address"
        />
        <label htmlFor="guest">UPS (next day air) $17.25</label>
      </div>
      <div className={`${CN}__radiobtn`}>
        <input
          className={`${CN}__radiobtn-input`}
          type="radio"
          name="ship-to-this-address"
          value="ship-to-different-address"
        />
        <label htmlFor="guest">UPS (second day air) $12.25</label>
      </div>
    </form>
    <Button className="black-button">continue</Button>
  </div>
);

export default ShippingMethod;
