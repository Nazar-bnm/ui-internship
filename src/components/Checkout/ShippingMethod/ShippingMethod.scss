@import '../../../scss/global_styles';

.shipping-method {
  @include flex($direction: column, $justify: space-between);
  padding: 20px;
  height: 200px;
  font-family: font(primary-light);
  text-transform: uppercase;

  &__radiobtn {
    margin: 5px 0;

    &-input {
      margin-right: 5px;
    }
  }
}