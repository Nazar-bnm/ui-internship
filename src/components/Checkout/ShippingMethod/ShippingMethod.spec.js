import React from 'react';
import { shallow } from 'enzyme';

import ShippingMethod from './ShippingMethod';

describe('<ShippingMethod />', () => {
  let wrapper = shallow(<ShippingMethod />);
  
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
