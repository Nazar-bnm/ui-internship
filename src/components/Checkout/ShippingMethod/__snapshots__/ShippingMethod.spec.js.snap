// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingMethod /> should match snapshot 1`] = `
<div
  className="shipping-method"
>
  <p
    className="shipping-method__text"
  >
    Please choose a shipping method to deliver your order:
  </p>
  <form
    className="shipping-method__radiobtns-form"
  >
    <div
      className="shipping-method__radiobtn"
    >
      <input
        className="shipping-method__radiobtn-input"
        name="ship-to-this-address"
        type="radio"
        value="ship-to-this-address"
      />
      <label
        htmlFor="ship-to-this-address"
      >
        UPS (ground) $7.25
      </label>
    </div>
    <div
      className="shipping-method__radiobtn"
    >
      <input
        className="shipping-method__radiobtn-input"
        name="ship-to-this-address"
        type="radio"
        value="ship-to-different-address"
      />
      <label
        htmlFor="guest"
      >
        UPS (7 days select) $9.75
      </label>
    </div>
    <div
      className="shipping-method__radiobtn"
    >
      <input
        className="shipping-method__radiobtn-input"
        name="ship-to-this-address"
        type="radio"
        value="ship-to-different-address"
      />
      <label
        htmlFor="guest"
      >
        UPS (next day air) $17.25
      </label>
    </div>
    <div
      className="shipping-method__radiobtn"
    >
      <input
        className="shipping-method__radiobtn-input"
        name="ship-to-this-address"
        type="radio"
        value="ship-to-different-address"
      />
      <label
        htmlFor="guest"
      >
        UPS (second day air) $12.25
      </label>
    </div>
  </form>
  <Button
    customClass="black-button"
    icon={null}
    onClick={[Function]}
  >
    continue
  </Button>
</div>
`;
